src/Claim.ts

```diff
@@ -57,20 +68,24 @@ export function createClaim(privateKey: string, type: ClaimType, attributes: Cla
     dateCreated: new Date(),
     attributes,
   }
-  const id = getClaimId(claim)
-  const signature = getClaimSignature(
-    {
+  const id = await getClaimId(claim)
```

Comment:
Isn't the id a multihash of the claim?

If it is, don't we just need to sign the id, and not the whole claim + id?

Reply:
> Does this one PR resolve the whole context of frost-api: 243?

>Canonicalize claims for hashing in Frost API #243

I realized as I built out my lengthy todo list, that most of it was outside the scope of the ticket as written.

We need to capture everything else, but this PR does do what the ticket is asking. (Once I modify as requested.

Reply:
It is just signing the id. However before that happens there is a fair amount of validation that occurs and requires the claim itself. Refer to line 46 of this file. Validations include:

* Is claim.publicKey present?
* Does the provided private key prove to verify the against the provided public key?
* Is clam.id present?
* Does a generated claim id match claim.id?

I didn't want to break this out without a lot more thought.

Reply:
> Isn't the id a multihash of the claim?

Now that you mention it, no, it isn't, but it shouldn't be too complicated to make it so. I've used the multihash lib and it's straight forward. 

Do we want to make the id a multihash (a breaking change) before mainnet beta? @ericelliott (We'd still address it in a separate PR)

Reply:
Let's consider this carefully. Is it possible two have two identical valid claims?

Put another way, would it ever make sense for a single entity to say the same thing about the same subject at a time such that the two identical claims have some kind of cumulative effect?